src/test/java/seedu/address/logic/graph/DfsHopCroftKarpTest.java

```diff
@@ -0,0 +1,53 @@
+package seedu.address.logic.graph;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.testutil.SampleGraph;
+import seedu.address.testutil.TestUtil;
+
+class DfsHopCroftKarpTest {
+    /**
+     * Integration testing of bfs and dfs.
+     */
+    @Test
+    public void execute_sampleGraph1After1Iteration_success() {
+        BipartiteGraph subjectGraph = SampleGraph.getSampleGraphOne();
+        int numInterviewees = subjectGraph.getNumInterviewees();
+        int numSlots = subjectGraph.getNumInterviewSlots();
+
+        List<InterviewSlotVertex> intervieweePredecessor = Arrays.asList(new InterviewSlotVertex[numInterviewees]);
+        List<List<IntervieweeVertex>> interviewSlotPredecessors = new ArrayList<>(numSlots);
+        boolean[] isUsedInterviewee = new boolean[numInterviewees];
+        boolean[] isUsedSlot = new boolean[numSlots];
+        TestUtil.fillWithSubLists(interviewSlotPredecessors, numSlots);
+
+        List<InterviewSlotVertex> lastLayer = new BfsHopCroftKarp(subjectGraph).execute(intervieweePredecessor,
+            interviewSlotPredecessors);
+        new DfsHopCroftKarp(subjectGraph).execute(lastLayer, intervieweePredecessor, interviewSlotPredecessors,
```

Comment:
I think it is better to break this up into 2 lines, one for initialising the `DfsHopCroftKarp` object and the next line to execute.